src/Discord.Net.Core/Cache/CacheableEntityExtensions.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Discord
+{
+    internal static class CacheableEntityExtensions
```

Comment:
Is there any reason why these methods arent returning the generic type parameters? We could downcast it later.

Reply:
Not really, its just whats there atm. it could change once I implement the rest of the entities